### PR TITLE
CSS Anatomy, resolves 525

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/css/concepts/anatomy/anatomy.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+Title: 'Anatomy'
+Description: 'There are two syntaxes used to apply CSS to a web-page. The first is as a _ruleset_, the second is as an _inline style_. A ruleset is defined within a block inside an HTML document, or in a separate file (called a stylesheet), and an inline style is applied as an attribute of the element the style is being applied to.'
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:3]:
```suggestion
Description: `The anatomy of CSS syntax spans between a rulesets in a stylesheet and inline styles within the markup.`
```

File: content/css/concepts/anatomy/anatomy.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Tags:
+  - 'DOM'
+  - 'Elements'
+  - 'Tags'
+CatalogContent:
+  - 'learn-css'
+  - 'paths/front-end-engineer-career-path'
+---
+
+There are two syntaxes used to apply CSS to a web-page. The first is as a _ruleset_, the second is as an _inline style_. A ruleset is defined within a block inside an HTML document, or in a separate file (called a stylesheet), and an inline style is applied as an attribute of the element the style is being applied to. The main difference between the two other than the syntax itself, is that CSS applied as an inline style only affects the one element it's applied to, whereas a CSS style defined in a ruleset can affect many elements at once. Also a ruleset can have many different styles defined, where an inline style can have only one.
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:16]:
```suggestion
There are two syntaxes for the anatomy of [CSS](https://www.codecademy.com/resources/docs/css) styles that are applied to a web-page:

* A _ruleset_ that is defined within either an [HTML](https://www.codecademy.com/resources/docs/html) document or in a separate file (known as stylesheets) that uses the **.css** extension.

* _Inline styles_ can be applied [attributes](https://www.codecademy.com/resources/docs/html/attributes) for styling [HTML elements](https://www.codecademy.com/resources/docs/html/elements) directly.

While inline styles are directly applied to a single element, rulesets can define multiple styles across multiple elements. Another key difference between the two is that a ruleset can utilize `id` and `class` [selectors](https://www.codecademy.com/resources/docs/css/selectors) for an improved styling experience.
```

File: content/css/concepts/anatomy/anatomy.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+CSS defined in a ruleset can also target elements in multiple different ways:
+
+- Element type
+- Class of the element
+- ID of the element
+- Parent of the Element
+
+Typically an inline style is used mainly for unique situations in HTML code where the style won't be repeated, in cases where HTML is dynamically generated and the CSS stylesheet is static, or in other situations where it isn't advisable (or possible) to edit the stylesheet(s) for a page.
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:25]:
```suggestion
Inline styles are commonly used for situations such as the following:

* The style being applied in the markup is unique and won't be repeated.
* The HTML markup is dynamically generated and the CSS stylesheet is static.
* Other situations where it isn't advisable (or possible) to edit the stylesheet(s) for a page.
```

File: content/css/concepts/anatomy/anatomy.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## Syntax
+
+Stylesheet/ruleset syntax (this will affect every `<p>` element in the HTML this apples to):
+
+```css
+p {
+  color: blue;
+}
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:34]:
```suggestion
/* Start of Ruleset */
selector:pseudo-class::pseudo-element {

  /* Declaration */
  property: value-or-keyword;

}
/* End of Ruleset */
```

File: content/css/concepts/anatomy/anatomy.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+- ID of the element
+- Parent of the Element
+
+Typically an inline style is used mainly for unique situations in HTML code where the style won't be repeated, in cases where HTML is dynamically generated and the CSS stylesheet is static, or in other situations where it isn't advisable (or possible) to edit the stylesheet(s) for a page.
+
+## Syntax
+
+Stylesheet/ruleset syntax (this will affect every `<p>` element in the HTML this apples to):
+
+```css
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:31]:
```suggestion
```pseudo
```

File: content/css/concepts/anatomy/anatomy.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+p {
+  color: blue;
+}
+```
+
+Inline style syntax (this only affects this single `<p>` element):
+
+```html
+<p style="color:blue;">Hello World!</p>
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:40]:
```suggestion
<element style="property:value;">Hello World!</element>
```

File: content/css/concepts/anatomy/anatomy.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+```css
+p {
+  color: blue;
+}
+```
+
+Inline style syntax (this only affects this single `<p>` element):
+
+```html
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:39]:
```suggestion
```pseudo
```

File: content/css/concepts/anatomy/anatomy.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+- Element type
+- Class of the element
+- ID of the element
+- Parent of the Element
+
+Typically an inline style is used mainly for unique situations in HTML code where the style won't be repeated, in cases where HTML is dynamically generated and the CSS stylesheet is static, or in other situations where it isn't advisable (or possible) to edit the stylesheet(s) for a page.
+
+## Syntax
+
+Stylesheet/ruleset syntax (this will affect every `<p>` element in the HTML this apples to):
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:29]:
```suggestion
Rulesets use the following syntax for selecting and styling any and every instance of an existing element:
```

File: content/css/concepts/anatomy/anatomy.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+```css
+p {
+  color: blue;
+}
+```
+
+Inline style syntax (this only affects this single `<p>` element):
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:37]:
```suggestion
Conversely, inline styles apply a visually-similar syntax to a single element:
```

File: content/css/concepts/anatomy/anatomy.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+## Anatomy
+
+There are several parts to the CSS in the above two examples.
+
+![Ruleset Anatomy](https://raw.githubusercontent.com/SSwiniarski/docs/css-anatomy/media/css-anatomy-1.png)
+
+![Inline Style Anatomy](https://raw.githubusercontent.com/SSwiniarski/docs/css-anatomy/media/css-anatomy-2.png)
+
+In the ruleset example, the initial `p` is known as a selector. In its simplest form, the selector can just be the element tag it applies to, i.e. `img`, `div`, `span`, `p`, though the selector can be much more specific about the elements it targets.
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:51]:
```suggestion
In the ruleset example, the initial `p` is a selector that can is applied to any and every valid instance of that element. Selectors possess a wide range of specificity, including:

* General selection by HTML element-name.
* Specific `class` and `id` selectors for elements.
* Pseudo-class and pseudo-element selectors.
* Reserved keyword rules like `!important` and `@media`.
```

File: content/css/concepts/anatomy/anatomy.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+Next in the ruleset syntax is the block of declarations between the two braces `{...}`. This is the block of code that defines the style that will be applied. In the inline style example, this is equivalent to the `style` attribute value.
+
+The declaration block, or attribute value, is made up of one or more declarations. In the examples, `color:blue;` is a declaration. A declaration is made up of a CSS property name, followed by a colon (`color:`) then a value, followed by a semicolon, (`blue;`). In either syntax, there can be an arbitrary number of declarations. Whitespace does not affect the declarations, but typically ruleset declarations are set off on their own lines, while inline style declarations are all placed on the same line.
```
Comment (Dusch4593) [inline content/css/concepts/anatomy/anatomy.md:55]:
```suggestion
The declaration block, or attribute value, is made up of one or more declarations. In the examples, `color:blue;` is a declaration. A declaration is made up of a CSS property name, followed by a colon (`color:`) then a value, followed by a semicolon, (`blue;`).

For either syntax, there can be an arbitrary number of declarations. While whitespacing has no effect, ruleset declarations are usually set on their own lines while inline style declarations are all placed on the same line.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Finished with my review and ready for you to take a look for more changes. 😃

**Comment by SSwiniarski** [comment]
@Dusch4593, I applied suggestions and made a few more revisions.

File: content/css/concepts/anatomy/anatomy.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+## Anatomy
+
+There are several parts to the CSS in both the ruleset and the inline-style:
+
+![Ruleset Anatomy](https://raw.githubusercontent.com/Codecademy/docs/main/media/css-anatomy-1.png)
+
+![Inline Style Anatomy](https://raw.githubusercontent.com/Codecademy/docs/main/media/css-anatomy-2.png)
+
+In the ruleset example above, the initial `p` is a selector that can is applied to any and every valid instance of that element. Selectors possess a wide range of specificity, including:
```
Comment (KTom101) [inline content/css/concepts/anatomy/anatomy.md:59]:
```suggestion
In the ruleset example above, the initial `p` is a selector that can be applied to any and every valid instance of that element. Selectors possess a wide range of specificity, including:
```

File: content/css/concepts/anatomy/anatomy.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+- General selection by HTML element-name.
+- Specific `class` and `id` selectors for elements.
+- Pseudo-class and pseudo-element selectors.
+- Reserved keyword rules like `!important` and `@media`.
+
+Next in the ruleset syntax is the block of declarations between the two braces `{...}`. This is the block of code that defines the style that will be applied. This is equivalent to the `style` attribute value in the inline-style example.
+
+The declaration block, or attribute value, is made up of one or more declarations. In the examples, `color:blue;` is a declaration. A declaration is made up of a CSS property name, followed by a colon (`color:`) then a value, followed by a semicolon, (`blue;`).
+
+For either syntax, there can be an arbitrary number of declarations. While whitespacing has no effect, ruleset declarations are usually set on their own lines while inline style declarations are all placed on the same line.
```
Comment (KTom101) [inline content/css/concepts/anatomy/anatomy.md:70]:
```suggestion
For either syntax, there can be an arbitrary number of declarations. While whitespace has no effect, ruleset declarations are usually set on their own lines while inline style declarations are all placed on the same line.
```

**Review by KTom101 — APPROVED** [review]
@SSwiniarski This looks good, just two minor changes!

**Comment by SSwiniarski** [comment]
@KTom101, done and done.

**Comment by KTom101** [comment]
@SSwiniarski We have to remove the italics on lines 18 & 20. @yangc95 brought it to @Dusch4593 and my attention that the product team doesn't want italics in the docs. Did you want to do something else instead or would just removing the italics be fine with you?

**Comment by SSwiniarski** [comment]
@KTom101. It still works without the italics. I pondered using quotes to set off the words, but I think just rewriting the two lines would work. I'll tweak it.

**Comment by KTom101** [comment]
@SSwiniarski Great! I think this is all good to merge if it's good with everyone else!

**Comment by yangc95** [comment]
@SSwiniarski I made some very minor changes to the visuals like removing extra spacing. I'm wondering if I could replace the images with the ones here:

Screen Shot 2022-02-22 at 3 30 51 PM
Screen Shot 2022-02-22 at 3 32 48 PM

In any case, this entry looks ready to go @KTom101 @Dusch4593

**Review by yangc95 — APPROVED** [review]

**Comment by SSwiniarski** [comment]
@yangc95, I updated the graphics, so we should be good to go.

**Comment by Dusch4593** [comment]
Nice work, everyone! 🙌  The changes are now live: https://www.codecademy.com/resources/docs/css/anatomy

**Pull request merged**